### DISCUSSION TITLE
Accessing files from and to Resolve

### DISCUSSION THREAD

**Discussion by ghost** [fat-tire/resolve#53] [Q&A] [answered]
Hello everyone,

Thank you for the this project, worked like a charm that first try even on Da Vinci Resolve 19 beta !

I'm having a slight problem with accessing my files from videos in my home directory inside the mount/home of the podman. Am i expected to create a symbolic link/pointer to my external home drive to the internal podman ? 

In essence i would like to access my files from and to (internally) the Da Vinci resolve podman. 

Any help would be greatly appreciated ! 
Thanks !

Kind Regards,
Tim

**Comment by fat-tire** [comment]
Yes, so one of the ideas and advantages with the container is to keep your personal stuff separate from what the resolve container can "see"-- but if you want to have the container to be able to see and interact with your host (the outside non-container)'s directory, you can override the normal behavior.

There are a lot of ways to do this, but maybe one of the easiest ways is to create a a file and call it `resolve.rc`.  in that file you'd put two lines:

RESOLVE_BIND_SOURCES=(<path_to_your_home_directory>)
RESOLVE_BIND_TARGETS=("/home/resolve/my_home_directory")

If the path to your home directory has spaces in it, you might want to put it in quotes. If you only want to give the Resolve container access to a directory, you can point to that instead of your whole home folder.

Now run `./resolve.sh` and in `/home/resolve` add a "mount point" like this:

`mkdir /home/resolve/my_home_directory`

You're all set up!  Now when you run `./resolve.sh` just do this first:

`$ export  RESOLVE_RC_PATH=/path/to/resolve.rc`

(you can also put the export line in a `.bashrc` or `.zshrc` file if you want it automatically run every time you open a shell terminal)

[See here for more explanation](https://github.com/fat-tire/resolve?tab=readme-ov-file#the-resolve_-environment-variables) and details of what you can do with this `resolve.rc` file.  It's pretty powerful because you can set a lot of stuff to be set up every time.

So inside the container, your "home directory" (or whatever directories you like) can be accessible from inside the container!

Does this make sense?  Last note-- you can also run it on a single line without the `export` command if you want like this:

`$ RESOLVE_RC_PATH=/path/to/resolve.rc /path/to/resolve.sh`

You can also add it to a desktop shortcut line as described in the link above:

` Exec=bash -c "cd $HOME/container/resolve && env RESOLVE_RC_PATH=$HOME/container/resolve/resolve.rc ./resolve.sh"`

This will let you have a clickable icon in the dock in GNOME.

**Reply by ghost** [reply]
```
# resolve.rc

RESOLVE_NETWORK="host" # Connects to internet

RESOLVE_LICENSES_AGREE="Y" # Agrees to licences

RESOLVE_BIND_SOURCES=("/home/****/Videos")
RESOLVE_BIND_TARGETS=("/home/****/Documents/containers/resolve/mounts/resolve-home") ← I suspect i have set this up wrong

# add any other configurations or commands here

echo "setup complete!" 
```

**Thanks for the prompt response !** 

I edited my .rc file (as shown) to what you recommended, which also gave me access to the internet!
With regards to your second point, i don't really quite get it. 

Am i expected to edit the mount resolve file to include a new home directory.

> Now run ./resolve.sh and in /home/resolve add a "mount point" like this:
> 
> mkdir /home/resolve/my_home_directory
> 
> You're all set up! Now when you run ./resolve.sh just do this first:
> 
> $ export  RESOLVE_RC_PATH=/path/to/resolve.rc

Sorry for any inconveniences caused, i am somewhat new to Linux

Kind Regards,
Tim

**Comment by fat-tire** [comment, marked as answer]
Oh it should look a bit more like this:

```bash
RESOLVE_BIND_SOURCES=("/home/****/Videos")  # this is the actual directory on the host (your regular computer)
RESOLVE_BIND_TARGETS=("/home/resolve/Videos") # this is the directory where it will apepar inside the container
```

What I meant by the part you quoted was that to match up these directories, the one on the container has to actually exist.  You can create it from inside the container or outside.  If inside, you'd do it like this:
```bash
$ ./resolve.sh /bin/bash  # to run the container in "shell mode" from wherever you put resolve.sh
bash-4.4$ # now you're in the container
bash-4.4$ mkdir /home/resolve/Videos # to create the directory where the host directory will appear
bash-4.4$ exit  # to quit the container
```

Alternatively you can create the "mount point" directory from outside the container by going to the mount directory and making it:
```bash
$ mkdir <path_to_resolve_repository>/mounts/resolve-home/Videos
```
So if you've put the resolve.sh script in `/home/my_account/containers/resolve/`, it would look something like:

```
$ mkdir /home/my_account/containers/resolve/mounts/resolve-home/Videos
```

Now that you've created an actual directory inside the container for your host's `Videos` directory to connect to, you should be able to run using the `resolve.rc` configuration as described previously, and everything should hopefully work.

For this example I chose to mount your `Videos` directory to `/home/resolve/Videos`, but you can map it to wherever you like.

**Reply by ghost** [reply]
Thank you very much this solved it ! 

Once again thank you for the prompt response !

**Comment by fat-tire** [comment]
I should add that you can do this as many times as you want with as many directories as you want by adding directories to `RESOLVE_BIND_SOURCES` and `RESOLVE_BIND_TARGETS` in pairs to `resolve.rc`, separated by spaces (not commas).
